conditions/operator_conditions.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conditions
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	api "github.com/operator-framework/api/pkg/operators/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var (
+	// ErrNoOperatorCondition indicates that the operator condition CRD is nil
+	ErrNoOperatorCondition = fmt.Errorf("operator Condition CRD is nil")
+)
+
+// TODO: verify from OLM if this will be the name of the environment variable
+// which is set for the Condition resource owned by the operator.
+const operatorCondEnvVar = "OPERATOR_CONDITION_NAME"
+
+var readNamespace = func() ([]byte, error) {
+	return ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+}
+
+// GetNamespacedName returns the NamespacedName of the CR. It returns an error
+// when the name of the CR cannot be found from the environment variable set by
+// OLM, or when the namespace cannot be found from the associated service account
+// secret.
+func GetNamespacedName() (*types.NamespacedName, error) {
+	conditionName := os.Getenv(operatorCondEnvVar)
+	if conditionName == "" {
+		return nil, fmt.Errorf("required env %s not set, cannot find operator condition CR for the operator", operatorCondEnvVar)
+	}
+	operatorNs, err := getOperatorNamespace()
```

Comment:
This code is duplicated in the [leader](https://github.com/operator-framework/operator-lib/blob/main/leader/leader.go#L240) package. Can you pull this out into a top-level internal package?

Additionally Operators that use this code won’t be able to run locally unless the environment creates the SA namespace file on each run. Unlike the leader package, which shouldn’t be executed, conditions should be set when running locally, so there needs to be a way to set this namespace. I suggest an environmental variable like `OPERATOR_NAMESPACE_LOCAL` that gets read when `getOperatorNamespace()` returns `ErrNoNamespace`